addons/account_edi/models/account_edi_format.py

```diff
@@ -506,21 +506,90 @@ def _retrieve_partner(self, name=None, phone=None, mail=None, vat=None):
         :param vat:     The vat number of the partner.
         :returns:       A partner or an empty recordset if not found.
         '''
-        domains = []
-        for value, domain in (
-            (name, [('name', 'ilike', name)]),
-            (phone, expression.OR([[('phone', '=', phone)], [('mobile', '=', phone)]])),
-            (mail, [('email', '=', mail)]),
-            (vat, [('vat', 'like', vat)]),
-        ):
-            if value is not None:
-                domains.append(domain)
-
-        domain = expression.AND([
-            expression.OR(domains),
-            [('company_id', 'in', [False, self.env.company.id])],
-        ])
-        return self.env['res.partner'].search(domain, limit=1)
+        def search_with_vat(extra_domain):
+            if not vat:
+                return None
+
+            # Sometimes, the vat is specified with some whitespaces.
+            normalized_vat = vat.replace(' ', '')
```

Comment:
What if it is defined with white spaces in the db? Some countries want that. Even if we decide to store the compacted version in db, it is not the case yet (and is not that trivial to do)

Reply:
If some people are encoding the VAT with whitespaces, this is not supported by this code, indeed.

Reply:
Then you broke it for all Swiss customers.

https://github.com/odoo/odoo/blob/68cdf057a9e769071fab6726eaf3b37d16959bf1/addons/base_vat/models/res_partner.py#L535-L537